lib/elixir/lib/string.ex

```diff
@@ -2231,6 +2231,10 @@ defmodule String do
   Returns the grapheme at the `position` of the given UTF-8 `string`.
   If `position` is greater than `string` length, then it returns `nil`.
 
+  Note: keep in mind this function has to linearly traverse the string.
+  If you want to access a string or a binary in constant time based on the
+  number of bytes, use `Kernel.binary_slice/3` or `:binary.at/2` instead.
```

Comment:
```suggestion
  > #### Linear Access {: .warning}
  >
  > This function has to linearly traverse the string.
  > If you want to access a string or a binary in constant time based on the
  > number of bytes, use `Kernel.binary_slice/3` or `:binary.at/2` instead.
```

Reply:
I agree this feels like a good use case.
I should also update https://hexdocs.pm/elixir/String.html#split_at/2 I think.